src/Externals/spire/arc-ball/ArcBall.cpp

```diff
@@ -76,7 +76,13 @@ void ArcBall::setLocationOnSphere(glm::vec3 location, glm::vec3 up)
 //------------------------------------------------------------------------------
 glm::quat ArcBall::quatFromUnitSphere(const glm::vec3& from, const glm::vec3& to)
 {
-  glm::vec3 axis = glm::normalize(glm::cross(from, to));
+  glm::vec3 cross = glm::cross(from, to);
+  glm::vec3 axis = glm::normalize(cross);
+
+  // Give arbitrary non-zero vector because no rotation
+  if (glm::length(cross) == 0.0)
```

Comment:
Shouldn't this have some tolerance?

Reply:
I was thinking the same thing

Reply:
I don't think it would have been a problem because cross products won't get small enough since it is based on pixel positions. I changed it to check nan instead so it should be more robust without the arbitrary threshold.